### ISSUE TITLE
defaults = _chart["defaults"].defaults is undefined

### ISSUE THREAD

**Issue by DiegoCampaneroNegrete** [reactchartjs/react-chartjs-2#639]
Hi and greattings, i maked a project in cube.js and afther that a "dashboard-app" of type 'real time dashboard' everything is ok until I run  the app and response whit that message:

`TypeError: _chart.default is undefined`

and i could't advanced afther that.

so i goes  to the file
 `node_modules/react-chartjs-2/es/index.js:644`
and I make a log of  the variable '_chart' and I see that only exist the field "_chart["default"]"

so i fix the error whit this: 

```javascript
// I change this:
// var defaults = _chart["default"].defaults;
// For this:
var defaults = _chart["defaults"];
```

I hope someone can fix this or use that issue for help

**Comment by SuspiciousLookingOwl** [comment]
Duplicate of  #635 and #634

TL;DR: use chart.js v2

```
npm i chart.js@2.9.4
```

**Comment by kapso** [comment]
Curious about any plans of supporting  chart.js v3 ?

**Comment by yuan9090** [comment]
@kapso According to https://github.com/reactchartjs/react-chartjs-2/issues/634#issuecomment-815160052 , maintainer do not maintain.

But I fork the repository, and merge [#599 3.x Migration](https://github.com/reactchartjs/react-chartjs-2/pull/599).
It compatible with Chart.js 3, you can try this: [@iftek/react-chartjs-3](https://github.com/yuan9090/react-chartjs-3)

**Comment by mehdi2010** [comment]
this problem is from chart.js package  just follow  below steps for resolve it
1:npm uninstall chart.js
2:npm i chart.js@2.3

**Comment by Yuhannakapali** [comment]
as the readme.md suggest chartjs@2.5 is recomended this issue occurs when you install chartjs@^3 follow following steps to solve this issue. 
1. npm uninstall chat.js || yarn remove chat.js
2. npm i chart.js@2.9 || yarn add chart.js@2.9

**Comment by prompt-07** [comment]
@Yuhannakapali  thanks it worked

**Comment by jtremback** [comment]
Might be good to put this requirement on 2.9.4 in the readme. Right now, when I come to this repo and try out the library, it is broken out of the box

**Comment by jtremback** [comment]
Using chart.js 2.9.4 does not seem to fix this.

Here are my dependencies:
```
    "chart.js": "2.9.4",
    "react-chartjs-2": "^2.11.1",
```
I have also verified that 2.9.4 is the version installed in my node_modules folder.

Here is my code:

```
import React from "react";
import { Line } from "react-chartjs-2";

const data = {
  labels: ["January", "February", "March", "April", "May", "June", "July"],
  datasets: [
    {
      label: "My First dataset",
      fill: false,
      lineTension: 0.1,
      backgroundColor: "rgba(75,192,192,0.4)",
      borderColor: "rgba(75,192,192,1)",
      borderCapStyle: "butt",
      borderDash: [],
      borderDashOffset: 0.0,
      borderJoinStyle: "miter",
      pointBorderColor: "rgba(75,192,192,1)",
      pointBackgroundColor: "#fff",
      pointBorderWidth: 1,
      pointHoverRadius: 5,
      pointHoverBackgroundColor: "rgba(75,192,192,1)",
      pointHoverBorderColor: "rgba(220,220,220,1)",
      pointHoverBorderWidth: 2,
      pointRadius: 1,
      pointHitRadius: 10,
      data: [65, 59, 80, 81, 56, 55, 40],
    },
  ],
};

export const ReactChartJs = () => {
  return (
    <div>
      <h2>Line Example</h2>
      <Line data={data} />
    </div>
  );
};
```

Here is the error:

```
TypeError: Cannot read property 'defaults' of undefined
(anonymous function)
node_modules/react-chartjs-2/es/index.js:643
  640 | }(_react["default"].Component);
  641 | 
  642 | exports.Scatter = Scatter;
> 643 | var defaults = _chart["default"].defaults;
  644 | exports.defaults = defaults;
```

**Comment by thebergamo** [comment]
Unfortunately I could only use it downgrading to chart.js@2.3 then it worked with latest version.

**Comment by jtremback** [comment]
I just used chart.js myself with a useEffect hook FWIW

**Comment by Neelam-Nishad** [comment]
> I just used chart.js myself with a useEffect hook FWIW

same issue with me

**Comment by YakovlevCoded** [comment]
#640

**Comment by duttanaman1** [comment]
TypeError: Cannot read property 'defaults' of undefined
(anonymous function)

 ```
 640 | }(_react["default"].Component);
  641 | 
  642 | exports.Scatter = Scatter;
> 643 | var defaults = _chart["default"].defaults;
  644 | exports.defaults = defaults;
```

I am using **"chart.js": "^2.9.4"**, and **"react-chartjs-2": "^2.11.1"**, still I am facing this issues. Please help, its urgent

**Comment by YakovlevCoded** [comment]
Hi! We have updated the wrapper. Try to use the latest version ^3.0.2 with Chart.js ^3.x.

**Comment by ashishsingh0660** [comment]
1st npm uninstall react-chartjs-2 chart.js 
Then use this command definitely solve your probleam

npm install --save react-chartjs-2 chart.js

**Comment by kapso** [comment]
@YakovlevCoded  

Tried the latest version but the following  `import { Bar, Chart, HorizontalBar } from 'react-chartjs-2';`

results in this error... `"export 'HorizontalBar' was not found in 'react-chartjs-2'`

these versions fail...
```
"chart.js": "3.1.1",
"react-chartjs-2": "3.0.2",
"react": "17.0.2",
"react-dom": "17.0.2",
```

the following versions work for us...
```
"chart.js": "2.9.4",
"react-chartjs-2": "2.11.1",
"react": "17.0.2",
"react-dom": "17.0.2",
```

**Comment by YakovlevCoded** [comment]
@kapsoHhorizontalBar chart type was removed. Horizontal bar charts can be configured using the new indexAxis option
https://www.chartjs.org/docs/latest/getting-started/v3-migration.html#chart-types
You can see example here: https://github.com/reactchartjs/react-chartjs-2/blob/master/example/src/charts/HorizontalBar.js

**Comment by dalelantowork** [comment]
Module not found: Can't resolve 'chart.js/auto' in '/Users/apple/Desktop/Netlify/navbar/client/node_modules/react-chartjs-2/dist' 

I get this error when installing an older version of reactjs-charts, i still could not make the defaults work

**Comment by YakovlevCoded** [comment]
@dalelantowork Hi! Try to update react-chartjs-2 and Chart.js to the latest version. If you want to use Chart.js 2.x version, use react-chartjs-2 2.11.1 version.

**Comment by vijaychouhan-rails** [comment]
@YakovlevCoded I am using latest version of both but getting same issue `defaults = _chart["defaults"].defaults`
